### DISCUSSION TITLE
Fully controlled table

### DISCUSSION THREAD

**Discussion by lkostrowski** [TanStack/table#2831] [Ideas]
I'm struggling second day to implement something like a Table with multiple point of control.

Eg.

1. My table works on its own (with filters, sorting)
2. I save table "state" in URL so page can be shared (I use `useControlledState` and serialize object. I use this object later for `initialState`. It causes some rendering problems but doesnt fail)
3. I need also external controls like filters, sorting, depending on context. Eg: there is a page with widgets and table. Clicking on widget should set some filters on table.

I have a problem to achieve that, I might be missing something with `stateReducer` (still figuring it out), but I wonder about a classic controlled component override (of course optional):

```
<Table state={externalState} useStateReducer={state => {
  // set some state in e.g. redux
  // dont return state, because its provided in different prop
}}  />
```

Does it make sense? Or maybe its achievable with another approach?

**Comment by tannerlinsley** [comment]
V7 just isn’t well suited for this. But v8 is being written with a much better controlled api.

**Reply by lkostrowski** [reply]
Thanks for answer. Is there a chance you guide me how to solve my problem (e.g access some internal instance setState etc?)

**Reply by tannerlinsley** [reply]
I wish there was an internal setState in v7, but there currently is not :(
